### DISCUSSION TITLE
Selenide 6.x still uses selenium 3.x

### DISCUSSION THREAD

**Discussion by jankosz** [selenide/selenide#1760] [Q&A] [answered]
I create small demo project with dependency only to selenide 6.3.5:

	<dependencies>
		<dependency>
			<groupId>com.codeborne</groupId>
			<artifactId>selenide</artifactId>
			<version>6.3.5</version>
		</dependency>
	</dependencies>

Inside selenide pom there is dependency to selenium-java 4.1.2:

    <dependency>
      <groupId>org.seleniumhq.selenium</groupId>
      <artifactId>selenium-java</artifactId>
      <version>4.1.2</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <artifactId>*</artifactId>
          <groupId>net.bytebuddy</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    
But when I run mvn dependency:tree I get following results:
com.example:demo:jar:0.0.1-SNAPSHOT
[INFO] \- com.codeborne:selenide:jar:6.3.5:compile
[INFO]    +- org.seleniumhq.selenium:selenium-java:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-api:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-chrome-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-edge-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-firefox-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-ie-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-opera-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-remote-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-safari-driver:jar:3.141.59:compile
[INFO]    |  +- org.seleniumhq.selenium:selenium-support:jar:3.141.59:compile
[INFO]    |  +- org.apache.commons:commons-exec:jar:1.3:compile
[INFO]    |  +- com.squareup.okhttp3:okhttp:jar:3.14.9:compile
[INFO]    |  \- com.squareup.okio:okio:jar:1.14.0:compile

Still selenium-java 3.141.59 is used.
I'm using maven 3.6.3
I tried removing selenide and selenium from local repository but with no luck 

At this point I'm not sure if this is issue with selenide or with maven.
Any ideas?

**Comment by asolntsev** [comment]
@jankosz obviously, you didn't show all your dependencies. Some of them forces Selenium 3.

Can you share the whole project or at least pom.xml?

**Comment by jankosz** [comment, marked as answer]
Hi @asolntsev,
You are right, I have more dependencies. As parent I use "spring-boot-starter-parent"
Here is my complete pom:

    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	    <modelVersion>4.0.0</modelVersion>
	    <parent>
		    <groupId>org.springframework.boot</groupId>
		    <artifactId>spring-boot-starter-parent</artifactId>
		    <version>2.6.5</version>
		    <relativePath/>
	    </parent>
	    <groupId>com.example</groupId>
	    <artifactId>demo</artifactId>
	    <version>0.0.1-SNAPSHOT</version>
	    <name>demo</name>
	    <description>Demo project for Spring Boot</description>
	    <properties>
		    <java.version>11</java.version>
	    </properties>
	    <dependencies>
		    <dependency>
			    <groupId>com.codeborne</groupId>
			    <artifactId>selenide</artifactId>
			    <version>6.3.5</version>
		    </dependency>
	    </dependencies>
    </project>

When I remove "spring-boot-starter-parent" parent, I'm getting correct 4.1.2 version of selenium-java.
Thanks for the hint!
Now I have 2 questions:
1. Why maven dependency:tree doesn't show where selenium-java: 3.141.59 is coming from?
2. I want to use spring boot libraries in my project. What is proper way to fix that? Explicitly state 4.1.2 version of selenium-java?

Thanks in advance.

**Reply by asolntsev** [reply]
Yes, spring boot starter forces the version of Selenium. 
Somehow it forces it so that we don't see in "dependency tree". :( 

But there is a solution:
```xml
<properties>
  <selenium.version>4.1.2</selenium.version>
</properties>
```

See https://rieckpil.de/write-concise-web-tests-with-selenide-for-java-projects/

**Comment by jankosz** [comment]
Thanks for your answer.
Unfortunately in our real project we don't use spring as a parent but rather we add spring boot dependencies in our dependencyManagement section:

	<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
		<modelVersion>4.0.0</modelVersion>
		<groupId>com.example</groupId>
		<artifactId>demo</artifactId>
		<version>0.0.1-SNAPSHOT</version>
		<name>demo</name>
		<description>Demo project for Spring Boot</description>
	
		<properties>
			<java.version>11</java.version>
			<selenium.version>4.1.2</selenium.version>
			<org.spring.boot.version>2.6.4</org.spring.boot.version>
		</properties>
	
		<dependencyManagement>
			<dependencies>
				<dependency>
					<groupId>org.springframework.boot</groupId>
					<artifactId>spring-boot-dependencies</artifactId>
					<version>${org.spring.boot.version}</version>
					<type>pom</type>
					<scope>import</scope>
				</dependency>
				<dependency>
					<groupId>com.codeborne</groupId>
					<artifactId>selenide</artifactId>
					<version>6.3.5</version>
				</dependency>
			</dependencies>
		</dependencyManagement>
	
		<dependencies>
			<dependency>
				<groupId>com.codeborne</groupId>
				<artifactId>selenide</artifactId>
				<version>6.3.5</version>
				<scope>test</scope>
			</dependency>
		</dependencies>
	</project>

And in such case overriding "selenium.version" property is not working (or am I doing sth wrong?)
Do you have recommendation how to deal with such scenario?

**Reply by asolntsev** [reply]
No, I don't. I am not a big fan of Spring.

**Reply by jankosz** [reply]
Ok, no worries. I guess this is more of a maven question.
I ended up adding explicit dependency to the selenium v4
Anyways, thanks for your help.
